ambry-rest/src/main/java/com.github.ambry.rest/Http2StreamHandler.java

```diff
@@ -31,21 +33,25 @@
   private NettyMetrics nettyMetrics;
   private NettyConfig nettyConfig;
   private PerformanceConfig performanceConfig;
+  private Http2ClientConfig http2ClientConfig;
   private RestRequestHandler requestHandler;
 
   public Http2StreamHandler(NettyMetrics nettyMetrics, NettyConfig nettyConfig, PerformanceConfig performanceConfig,
-      RestRequestHandler requestHandler) {
+      Http2ClientConfig http2ClientConfig, RestRequestHandler requestHandler) {
     this.nettyMetrics = nettyMetrics;
     this.nettyConfig = nettyConfig;
     this.performanceConfig = performanceConfig;
+    this.http2ClientConfig = http2ClientConfig;
     this.requestHandler = requestHandler;
   }
 
   @Override
   public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
     ctx.pipeline().addLast(new Http2StreamFrameToHttpObjectCodec(true));
+    // TODO: Remove HttpObjectAggregator. Currently, without HttpObjectAggregator, last httpContent can't be recognized as LastHttpContent.
```

Comment:
what do you want to remove it in the TODO? Keep HttpObjectAggregator seems fine to me.

Reply:
HttpObjectAggregator caused an extra copy and extra cpu resource.

I am actually thinking to remove NettyMessageProcessor. 

HttpObjectAggregator is reliable and simple.

Reply:
What would be the pathway to removing HttpObjectAggregator? is it a bug in the Netty http2 handlers or a bug in NettyMessageProcessor?

Reply:
It looks like a compatible issue between http2 impl and NettyMessageProcessor. 
I need to re-consider what to do next step. Two choices:
1. fix the bug 
2. do not use NettyMessageProcessor

Will fix this is follow up PRs.

Reply:
Yeah, the NettyMessageProcessor doesn't expect a FullHttpRequest, it expects different http components. If we can get rid of NettyMessageProcessor here, I would totaly vote for it. NettyMessageProcessor does look like over complicated processor that was designed for frontend.

Reply:
in the ambry server, we don't have multiple chunks, we don't have requests or response larger than 4MB. Lots of logic in the frontend just don't make too much sense in the ambry server.

Reply:
We do have requests and responses larger than 4 MB for replication batch-gets (and that period of time last year when we ran with 8 MB chunks), so there can be value from streaming. However, a lot of things, such as the way we prefetch blobs in the server, to get tangible benefit (other than abstract goals like sharing code btwn frontend/server) out of NettyMessageProcessor or similar pieces.